Sources/Moya/MoyaProvider+Internal.swift

```diff
@@ -83,18 +84,57 @@ public extension MoyaProvider {
             switch stubBehavior {
             case .never:
                 switch target.task {
-                case .request:
+                case .requestData(let data):
+                    preparedRequest.httpBody = data
+                    cancellableToken.innerCancellable = self.sendRequest(target, request: preparedRequest, queue: queue, progress: progress, completion: networkCompletion)
+
+                case let .requestEncoded(parameters: parameters, encoding: parameterEncoding):
+                    do {
```

Comment:
Usually this is done in an `Endpoint`, take a look [here](https://github.com/Moya/Moya/blob/master/Sources/Moya/MoyaProvider%2BDefaults.swift#L16-L22). If we cannot encode parameters, we throw a `MoyaError.requestMapping(_:)` error to the user.